test/vfd_swmr_dsetchks_writer.c

```diff
@@ -0,0 +1,2269 @@
+        TEST_ERROR;
+    }
+
+    /* Verify the data read in `rbuf` is as the fill value expected */
+    for (i = 0; i < block[0] * block[1]; i++) {
+        if (action == GWRITES || action == TWRITES) {
+            if (rbuf[i] != FILL_FULL) {
+                printf("Invalid value for dataset for GWRITES/TWRITES\n");
+                TEST_ERROR;
+            }
+        }
+        else {
+            HDassert(action == PWRITES || action == LWRITES);
+            if (rbuf[i] != FILL_PARTIAL) {
+                printf("Invalid value for dataset for GWRITES/TWRITES\n");
+                TEST_ERROR;
+            }
+        }
+    }
+
+    if (H5Sclose(sid) < 0) {
+        printf("H5Sclose failed\n");
+        TEST_ERROR;
+    }
+
+    if (rbuf)
+        HDfree(rbuf);
+
+    return true;
+
+error:
+    H5E_BEGIN_TRY
+    {
+        H5Sclose(sid);
+        H5Sclose(mem_sid);
+    }
+    H5E_END_TRY;
+
+    if (rbuf)
+        HDfree(rbuf);
+
+    return false;
+
+} /* verify_read_dset() */
+
+/*
+ * Verify the increase or decrease of dimenion sizes for the specified datasets.
+ */
+static bool
+verify_dsets_extent(unsigned action, const state_t *s, const dsets_state_t *ds, unsigned which)
+{
+    unsigned rows = s->rows;
+    unsigned cols = s->cols;
+
+    HDassert(action == INCR_EXT || action == DECR_EXT);
+    HDassert(s->fa_index || s->ea_index || s->bt2_index);
+
+    /* s->xincrs can be 0 or the increased extent of the dataset */
+    if (action == DECR_EXT) {
+        rows = s->rows + s->xincrs;
+        cols = s->cols + s->xincrs;
+    }
+
+    if (s->fa_index) {
+        dbgf(2, "Verify dataset extent for FA dataset\n");
+        if (!verify_dset_extent_real(action, ds->fa_did, rows, cols, which)) {
+            printf("verify_read_dset() to dataset with fixed array index failed\n");
+            TEST_ERROR;
+        }
+    }
+
+    if (s->ea_index) {
+        dbgf(2, "Verify dataset extent for EA dataset\n");
+        if (!verify_dset_extent_real(action, ds->fa_did, rows, cols, which)) {
+            printf("verify_read_dset() to dataset with fixed array index failed\n");
+            TEST_ERROR;
+        }
+    }
+
+    if (s->bt2_index) {
+        dbgf(2, "Verify dataset extent for BT2 dataset\n");
+        if (!verify_dset_extent_real(action, ds->bt2_did, rows, cols, which)) {
+            printf("verify_read_dset() to dataset with fixed array index failed\n");
+            TEST_ERROR;
+        }
+    }
+
+    return true;
+
+error:
+    return false;
+
+} /* verify_dsets_extent() */
+
+/*
+ * Do the real work of verifying the increase/decrease for the dataset dimension sizes
+ */
+static bool
+verify_dset_extent_real(unsigned action, hid_t did, unsigned rows, unsigned cols, unsigned which)
+{
+    hsize_t dims[2];
+    hid_t   sid = badhid;
+
+    /* Refresh the dataset */
+    if (H5Drefresh(did) < 0) {
+        printf("H5Drefresh dataset failed\n");
+        TEST_ERROR;
+    }
+
+    if ((sid = H5Dget_space(did)) < 0) {
```

Comment:
sid should be closed at the end of this routine.

Reply:
Corrected